### PR TITLE
[HomeKit] Update up to Xcode 9.2 Beta 2

### PR DESCRIPTION
**Pull request by dalexsoto**

### CODE REVIEW

File: src/HomeKit/HMEnums.cs [hunk 1]
```diff
@@ -1070,6 +1108,18 @@ public enum HMAccessoryCategoryType {
...
+		[Field ("HMAccessoryCategoryTypeSprinkler")]
+		Sprinkler,
+
+		[Watch (4,2), TV (11,2), iOS (11,2)]
+		[Field ("HMAccessoryCategoryTypeFaucet")]
+		Faucet,
+
+		[Watch (4,2), TV (11,2), iOS (11,2)]
+		[Field ("HMAccessoryCategoryTypeShowerHead")]
+		ShowerHead,
```
Comment (VincentDondain) [inline src/HomeKit/HMEnums.cs]: Reeeeeeeeeeally HomeKit shower heads... Ahem no comment (:

File: tests/introspection/ApiProtocolTest.cs [hunk 2]
```diff
@@ -122,6 +122,7 @@ protected virtual bool Skip (Type type, string protocolName)
 				case "NSMeasurement`1":
 				// iOS 11 / tvOS 11
 				case "VSSubscription":
+				case "HMAccessControl": // not in headers
```
Comment (VincentDondain) [inline tests/introspection/ApiProtocolTest.cs]: Minor but I guess you meant `Conformance not in headers`, this makes it sound like `HMAccessControl` is not available in the headers.

**Comment by dalexsoto** [comment]
build hang (known issue) https://github.com/xamarin/maccore/issues/572

File: tests/introspection/ApiProtocolTest.cs (hunk 2 above)
Comment (spouliot) [reply tests/introspection/ApiProtocolTest.cs]: should not be needed, it was added in the first PR (in iOSProtocolTest.cs)

File: tests/introspection/ApiProtocolTest.cs [hunk 3]
```diff
@@ -151,6 +152,7 @@ protected virtual bool Skip (Type type, string protocolName)
 				// xcode 9
 				case "NSConstraintConflict": // Conformance not in headers
 				case "VSSubscription":
+				case "HMAccessControl": // not in headers
```
Comment (spouliot) [inline tests/introspection/ApiProtocolTest.cs]: should not be needed, it was added in the first PR (in iOSProtocolTest.cs)

**Comment by dalexsoto** [comment]
@spouliot fixed, thanks!

File: src/HomeKit/HMEnums.cs (hunk 1 above)
Comment (rolfbjarne) [reply src/HomeKit/HMEnums.cs]: Ah, the joys of powered shower heads. There's a reason they're called [suicide showers](https://www.google.es/search?q=suicide+shower)...

**Pull request merged**